src/devices/machine/i8291a.h

```diff
@@ -0,0 +1,408 @@
+	{ return m_dreq_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_trig_write_cb(Object &&cb)
+	{ return m_trig_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_eoi_write_cb(Object &&cb)
+	{ return m_eoi_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_dav_write_cb(Object &&cb)
+	{ return m_dav_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_nrfd_write_cb(Object &&cb)
+	{ return m_nrfd_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_ndac_write_cb(Object &&cb)
+	{ return m_ndac_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_srq_write_cb(Object &&cb)
+	{ return m_srq_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_dio_write_cb(Object &&cb)
+	{ return m_dio_write_func.set_callback(std::forward<Object>(cb)); }
+
+	template <class Object> devcb_base& set_dio_read_cb(Object &&cb)
+	{ return m_dio_read_func.set_callback(std::forward<Object>(cb)); }
+
+	// Signal inputs
+	DECLARE_WRITE_LINE_MEMBER(reset_w);
+	DECLARE_WRITE_LINE_MEMBER(dack_w);
+
+	// GPIB port
+	DECLARE_WRITE_LINE_MEMBER(eoi_w);
+	DECLARE_WRITE_LINE_MEMBER(dav_w);
+	DECLARE_WRITE_LINE_MEMBER(nrfd_w);
+	DECLARE_WRITE_LINE_MEMBER(ndac_w);
+	DECLARE_WRITE_LINE_MEMBER(ifc_w);
+	DECLARE_WRITE_LINE_MEMBER(srq_w);
+	DECLARE_WRITE_LINE_MEMBER(atn_w);
+	DECLARE_WRITE_LINE_MEMBER(ren_w);
+	DECLARE_WRITE8_MEMBER(dio_w);
+
+	// register r/w functions
+
+	DECLARE_WRITE8_MEMBER(dout_w);
+	DECLARE_WRITE8_MEMBER(ie1_w);
+	DECLARE_WRITE8_MEMBER(ie2_w);
+	DECLARE_WRITE8_MEMBER(spoll_mode_w);
+	DECLARE_WRITE8_MEMBER(addr_mode_w);
+	DECLARE_WRITE8_MEMBER(aux_mode_w);
+	DECLARE_WRITE8_MEMBER(addr01_w);
+	DECLARE_WRITE8_MEMBER(eos_w);
+
+	DECLARE_READ8_MEMBER(din_r);
+	DECLARE_READ8_MEMBER(ints1_r);
+	DECLARE_READ8_MEMBER(ints2_r);
+	DECLARE_READ8_MEMBER(spoll_stat_r);
+	DECLARE_READ8_MEMBER(addr_stat_r);
+	DECLARE_READ8_MEMBER(cpt_r);
+	DECLARE_READ8_MEMBER(addr0_r);
+	DECLARE_READ8_MEMBER(addr1_r);
+	void map(address_map &map);
+
+private:
+
+	// signal output
+	devcb_write_line m_int_write_func;
+	devcb_write_line m_dreq_write_func;
+	devcb_write_line m_trig_write_func;
+	devcb_write_line m_eoi_write_func;
+	devcb_write_line m_dav_write_func;
+	devcb_write_line m_nrfd_write_func;
+	devcb_write_line m_ndac_write_func;
+	devcb_write_line m_srq_write_func;
+
+	devcb_write8 m_dio_write_func;
+	devcb_read8 m_dio_read_func;
+
+	void set_dav(bool state);
+	void set_nrfd(bool state);
+	void set_ndac(bool state);
+	void set_eoi(bool state);
+	void set_srq(bool state);
+
+
+	// device-level overrides
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	void update_int(void);
+	void handle_command(void);
```

Comment:
C++ doesn't support K&R calling convention - no need for `void` here.

Reply:
fixed